### ISSUE TITLE
Migrate "Customer Service > Merchandise Returns" Back Office section

### ISSUE THREAD

**Issue by matks** [PrestaShop/PrestaShop#24421]
## Context

Part of the [Symfony migration project](https://github.com/PrestaShop/PrestaShop/issues/14770).

The **"Customer Service > Merchandise Returns"** section of the Back Office covers
the full lifecycle of merchandise return requests (RMA) submitted by customers.
It consists of two pages and one settings form, all currently served by the legacy
`AdminReturnController` and pending full migration to Symfony/Twig.

> **Naming convention:** In the UI, this feature is called "Merchandise Returns".
> In the codebase (legacy and new), the canonical name is **OrderReturn**.
> All new classes, namespaces, routes and services introduced during this migration
> must use the `OrderReturn` naming. A dedicated refactoring task will align the
> remaining legacy-named files once the feature migration is complete.

<img width="3426" height="1387" alt="Image" src="https://github.com/user-attachments/assets/70e33d7b-9b13-49fd-a3d8-93d8566fbf77" />

---

### 1. Merchandise Returns — List page

**Legacy URL:** `?controller=AdminReturn`
**Migrated URL:** `/index.php/sell/customer-service/merchandise-return/`

Contains:
- A **grid listing** all merchandise return requests, with columns:
  ID, Order ID, Status, Date issued.
- A **settings form** ("Merchandise return (RMA) options") allowing the BO user
  to enable/disable returns and configure related options.

### 2. Merchandise Return — View & Edit page

**Legacy URL:** `?controller=AdminReturn&id_order_return={id}&vieworder_return`
**Migrated URL:** `/index.php/sell/customer-service/merchandise-return/{id}/edit`

Contains:
- A **summary header**: linked customer, linked order, current return state
  (editable via a dropdown + save action).
- A **products grid**: the list of returned products with their quantities
  and declared conditions.
- A **"Download PDF" action**: when the return is in "Waiting for package" state,
  a printable return label can be generated and downloaded.
**Warning**: In this story we'll not handle the PDF Generation refactory = we still use legacy PDF Generator. We'll create another story to handle this part.

---

## Acceptance criteria

- [ ] All pages are accessible via their Symfony URL and render correctly
      with the PrestaShop Twig layout (no legacy bridge / `@admin_legacy_cc`).
- [ ] The merchandise returns grid (list page) is fully functional:
      filtering, sorting, pagination, bulk actions.
- [ ] The RMA settings form saves correctly in all configurations
      (single store and multistore).
- [ ] The View & Edit page displays the full products grid.
- [ ] The "Download PDF" button is present and generates a valid PDF label
      when the return is in "Waiting for package" state.
- [ ] All new code uses the `OrderReturn` naming convention.
      All legacy `MerchandiseReturn`-named files are renamed accordingly.
- [ ] Functional QA sign-off on the complete Symfony flow.
- [ ] Following issue #27628 : we need also to display a modal when we click on delete for a product in the RMA
- [ ] Instead of removing directly the product related of RMA : we change the behaviour to delete the product when we click on "Save"

---

## Current migration status

| Component | Status | Reference |
|---|---|---|
| Feature flag Merchandise return | ✅ Merged | PR #15022 |
| Merchandise returns grid (list) | ✅ Merged | PR #15022 |
| RMA settings form (list page) | ✅ Merged (bugs reported) | PR #15022 |
| View & Edit page — state change | ✅ Merged | PR #28642 |
| View & Edit page — products grid | ⏳ Pending | — |
| View & Edit page — PDF download | ⏳ Pending | — |
| RMA settings form — bug fixes | ⏳ Pending | #19379 #19331 #27628 |
| Rename `MerchandiseReturn` → `OrderReturn` | ⏳ Pending | — |

---

### ✅ Completed

- [x] Migrate the merchandise returns listing and RMA settings form
      → PR #15022
- [x] Migrate the View & Edit page *(state change only)*
      → PR #28642
- [x] Add multistore behavior → #23610
- [x] Fix bug → #23605 · #23606 · #23609

### 🔧 Remaining

- [ ] Build the products grid inside the View & Edit page
- [ ] Build the "Download PDF" button and PDF generation in the View & Edit page
- [ ] Fix bugs in the RMA settings form → #19379 · #19331 · #27628
- [ ] Rename remaining `MerchandiseReturn` classes/files to `OrderReturn`
- [ ] Un-hide migrated pages and remove `AdminReturnController`
- [ ] Fix bug #19379
- [ ] Fix bug #19331 
- [ ] Anothers PRs are currently open on this topic - we can use it or not: https://github.com/PrestaShop/PrestaShop/pull/31013 / https://github.com/PrestaShop/PrestaShop/pull/32048
